openmarket-server/src/main/java/co/unicauca/openmarket/server/infra/OpenMarketHandler.java

```diff
@@ -0,0 +1,267 @@
+package co.unicauca.openmarket.server.infra;
+
+import co.unicauca.openmarket.commons.domain.Delivery;
+import co.unicauca.openmarket.commons.domain.Order;
+import co.unicauca.openmarket.commons.domain.Product;
+import co.unicauca.openmarket.commons.domain.ShoppingCart;
+import co.unicauca.openmarket.commons.domain.Category;
+import co.unicauca.openmarket.commons.domain.StateProduct;
+import co.unicauca.openmarket.commons.domain.User;
+import co.unicauca.openmarket.commons.infra.JsonError;
+import co.unicauca.openmarket.commons.infra.Protocol;
+import co.unicauca.openmarket.server.access.IProductRepository;
+import co.unicauca.openmarket.server.domain.service.OpenMarketFacade;
+import co.unicauca.strategyserver.infra.ServerHandler;
+
+import com.google.gson.Gson;
+
+import java.text.SimpleDateFormat;
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+/**
+ * @author Braian Rey / Arturo Restrepo Ruiz
+ */
+public class OpenMarketHandler {
+    private Map<String, Consumer<Protocol>> actionMap;
```

Comment:
en este hashamp en lugar de ser Consumer deberia ser Function ya que el Consumer en java solo te va ejecutar una acción pero esta no te va a devolver un valor, y en cada metodo se devuelve una String

Reply:
por lo tanto queda:  private Map<String, Function<Protocol, String>> actionMap;

Reply:
Si se usa ese entonces ya no seria necesario aplicar el this.reply = en cada request ya que ese si retorna